test/index.ts

```diff
@@ -702,6 +702,30 @@ test('schema - valid set', t => {
 	});
 });
 
+test('schema - valid set with full schema', t => {
+	const schema: Schema<{foo: {bar: number; foobar: number}}> = {
+		type: 'object',
+		properties: {
+			foo: {
+				type: 'object',
+				properties: {
+					bar: {
+						type: 'number'
+					},
+					foobar: {
+						type: 'number',
+						maximum: 100
+					}
+				}
+			}
+		}
```

Comment:
Can you include a top-level only property? Just to make sure it works.

Reply:
Added additional tests